src/lib/gossip_net/gossip_net.ml

```diff
@@ -639,9 +647,13 @@ module Make (Message : Message_intf) : S with type msg := Message.msg = struct
                     handle_protocol net2 ~on_handler_error:`Raise
                       ~protocol:disc_proto (fun stream ->
                         let _, w = Stream.pipes stream in
-                        let pushback = Pipe.write w
-                          ( Node_addrs_and_ports.to_peer config.addrs_and_ports
-                          |> Peer.to_yojson |> Yojson.Safe.to_string ) in Pipe.close w ; pushback )
+                        let pushback =
+                          Pipe.write w
+                            ( Node_addrs_and_ports.to_peer
+                                config.addrs_and_ports
+                            |> Peer.to_yojson |> Yojson.Safe.to_string )
+                        in
+                        Pipe.close w ; pushback )
```

Comment:
is this close safe? Does the close need to happen in `Deferred.upon` of the pushback?

Reply:
This is safe: it synchronously prevents new writes from entering the queue, without modifying the contents of the queue.